### DISCUSSION TITLE
Cursor randomly jumps when pressing tab

### DISCUSSION THREAD

**Discussion by ijsnow** [LazyVim/LazyVim#1985] [Q&A] [answered]
I have this weird issue where my cursor will randomly jump to what feels like a random location when I press tab. This doesn't happen each time I press tab so I've been ignoring it from some time but it would be great to finally get it to stop. I can't think of any plugins or configuration changes that I made that would cause this. Has anyone else experienced this? Any suggestions for debugging or narrowing down the issue would be appreciated! 

[Here is my configuration.](https://github.com/ijsnow/nvim) I use nvim inside of zellij with alacritty.

**Comment by cerebrl** [comment]
I've observed a similar behavior, and I _think_ it's from one of the snippets plugins. I have not taken the time to fully verify it, but my guess is that when I don't fill out a snippet completely, leaving some replacement text unchanged. The snippet plugin is still "listening" for the tab key to "jump" to that unfinished replacement, but the code or cursor is now "out of sync", so to speak.

Full disclosure: I'm a total n00b though, so feel free to ignore me :)

**Comment by lvl4ul2i** [comment]
It's the LuaSnip behavior , you can disable this by:
```lua
{
  "L3MON4D3/LuaSnip",
  keys = function()
    return {}
  end,
}
```
I think you are trying to use a snippet but you leave tabs waiting for input.
When you try to use the tab on another line it goes back to its place in the snippet.
If you want to use `<tab>` for completion and snippets (supertab) checkout the Recipes section on the docs. I personally like LuaSnip a lot.

**Reply by ijsnow** [reply]
Hmmm I’d like to use LuaSnip but not randomly jump around when not intended. I’ll take a look at the docs and see what I can do. Thanks!

**Reply by jyuan0** [reply]
Maybe take a look at `delete_check_events` in the LuaSnip config/options. I can't find the issue/discussion, but I remember something similar being described and the resolution was changing that option to `TextChanged` and/or `InsertLeave`.

**Reply by lvl4ul2i** [reply, marked as answer]
Try with 

```lua
  {
    "L3MON4D3/LuaSnip",
    opts = {
      history = true,
      region_check_events = "InsertEnter",
      delete_check_events = "TextChanged",
    },
  },
```

**Reply by ijsnow** [reply]
I've had this ^ as the configuration for a few days and haven't noticed the issue. Thanks!
